source/Ship.cpp

```diff
@@ -1435,6 +1436,37 @@ string Ship::GetHail(map<string, string> &&subs) const
 
 
 
+bool Ship::CanSendHail(const PlayerInfo &player, bool allowUntranslated) const
+{
+	const System *playerSystem = player.GetSystem();
+	if(!playerSystem)
+		return false;
+
+	// Make sure this ship is in the same system as the player.
+	if(GetSystem() != playerSystem)
+		return false;
+
+	// Player ships shouldn't send hails.
+	const Government *gov = GetGovernment();
+	if(!gov || IsYours())
+		return false;
+
+	// Make sure this ship is able to send a hail.
+	if(IsDisabled() || !Crew()
+			|| Cloaking() >= 1. || GetPersonality().IsMute())
```

Comment:
```suggestion
	if(IsDisabled() || !Crew() || Cloaking() >= 1. || GetPersonality().IsMute())
```
By my count, this line is exactly 80 characters so doesn't need to be split.

~~Also, `automated hail`s sent when a ship has no crew when?~~